### PR TITLE
UI improvement of "Request Access"

### PR DESCRIPTION
**Pull request by SofiaSazonova**
### Feature or Bugfix
- UI improvement

### Detail
- Dataset Page: for all users "Request access" button is added (pics 1, 2)
- Dataset List Page: "Learn more" removed (duplicated functionality), "Request access" added (pic 3)
- Catalog page: text added near the icon (pic 4)

### Relates
- #1008 

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by SofiaSazonova** [comment]
Screenshot 2024-04-29 at 14 15 28
Screenshot 2024-04-29 at 14 15 08
Screenshot 2024-04-29 at 14 01 10
Screenshot 2024-04-29 at 14 00 59

**Comment by SofiaSazonova** [comment]
@zsaltys have a look, please. Did I understand you correctly?
@dlpzx what do you think?

**Comment by dlpzx** [comment]
I agree with the improvements, my only comment would be the "Request access" button for dataset owners, i think in that case it is not necessary (so pic 2 yes, pic 1 no)

File: frontend/src/modules/S3_Datasets/views/DatasetView.js [hunk 1]
```diff
@@ -262,50 +274,74 @@ const DatasetView = () => {
...
-                </Box>
-              </Grid>
-            )}
+                )}
+                <RequestAccessModal
+                  onApply={handleRequestAccessModalClose}
+                  onClose={handleRequestAccessModalClose}
+                  open={isRequestAccessOpen}
+                  stopLoader={() => setIsOpeningModal(false)}
+                  hit={{ _id: dataset.datasetUri, resourceKind: 's3_datasets' }}
```
Comment (noah-paige) [inline frontend/src/modules/S3_Datasets/views/DatasetView.js:341]: should resourceKind be `dataset` here?

File: frontend/src/modules/S3_Datasets/components/DatasetListItem.js [hunk 2]
```diff
@@ -2,29 +2,42 @@ import {
   Box,
   Button,
   Card,
+  CircularProgress,
```
Comment (noah-paige) [inline frontend/src/modules/S3_Datasets/components/DatasetListItem.js:5]: I do not think it makes sense to have request access button + modal for dataset list page. For dataset list, only datasets which the current user's team is admin / steward to or already shared will show

It is true there are cases where this user then needs to share it to another team or a consumption role that they are a part of or manage respectively - but I think it is more straightforward to keep the `Learn More` from the List View 

And then have the added Request Access Button for all to see on the `DatasetView` page (as you have done so already)

**Review by noah-paige — APPROVED** [review]
tested and looks good!

**Pull request merged**